hazelcast/serialization/portable/writer.py

```diff
@@ -145,6 +194,23 @@ def write_utf(self, field_name, value):
     def write_utf_array(self, field_name, values):
         self.write_string_array(field_name, values)
 
+    def write_nullable_field(self, field_name: str, field_type: FieldType, value, func):
+        self._set_position(field_name, field_type)
+        is_nil = value is None
+        self._out.write_boolean(is_nil)
+        if not is_nil:
+            func(self._out, value)
+
+    def write_object_array_field(self, field_name, field_type, values, func):
+        self._set_position(field_name, field_type)
+        length = NULL_ARRAY_LENGTH if values is None else len(values)
+        if length < 0:
+            self._out.write_int(NULL_ARRAY_LENGTH)
+            return
+        self._out.write_int(length)
+        for i in values:
+            func(self._out, i)
```

Comment:
don't you need to set the item positions like java?

```
                int position = out.position();
                if (values[i] == null) {
                    throw new HazelcastSerializationException("Array items can not be null");
                } else {
                    out.writeInt(offset + i * INT_SIZE_IN_BYTES, position);
                    writer.write(out, values[i]);
                }
```

Reply:
The difference was due to porting the Go client code but I guess there is a different mechanism. Now converted to Java client version (also Node.js) at https://github.com/hazelcast/hazelcast-python-client/pull/569/commits/6b34418b0dd46c23cb7a5ed87e84d83865ce27e1. 

For comparison you can look at Node.js client: [DefaultPortableWriter.ts#L249](https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/src/serialization/portable/DefaultPortableWriter.ts#L249)

Reply:
Can you create a task for the Go repo then? Because, if they are not setting the positions, it won't be interoperable with Java, and that is a clear defect.

Reply:
@mehmettokgoz @mdumandag  This is the related Go client code, it seems to be almost the same with the Java client implementtion, am I missing something?
```
	innerOffset := pw.output.Position()
	pw.output.WriteZeroBytes(length * 4)
	for i := 0; i < length; i++ {
		sample := portableArray[i]
		posVal := pw.output.Position()
		pw.output.PWriteInt32(innerOffset+int32(i)*Int32SizeInBytes, posVal)
		pw.serializer.WriteObject(pw.output, sample)
	}
```
https://github.com/hazelcast/hazelcast-go-client/blob/77ae328a77da2bc6214f0a4d70c01dcf72858157/internal/serialization/default_portable_writer.go#L161

Reply:
This one seems OK, but we were talking about methods like writeTimeArray, writeDateArray. For such types, go seem to write the position only for the top level field, and skip writing positions for each array element.

Reply:
NVM, there is a separate writeObjectArray method which doesn't exist in the Go client, here's the relevant ticket: https://github.com/hazelcast/hazelcast-go-client/issues/894